File: PATTERNS/Pattern48/Pattern48.js
function printPattern(rows) {
    let col, num = 0;

    // Loop for each row
    for (let row = 1; row <= rows; row++) {

        // When row number is odd, print in increasing order
        if (row % 2 !== 0) {

            // Printing in ascending order
            for (col = num + 1; col < num + row; col++)
                process.stdout.write(col + " ");
            console.log(col);

            // Update value of 'num'
            num = col;
        }

        // When row number is even, print in descending order
        else {

            // Update value of 'num'
            num = num + row - 1;

            // Print numbers in decreasing order
            for (col = num; col > num - row + 1; col--)
                process.stdout.write(col + " ");
            console.log(col);
        }
    }
}

File: PATTERNS/Pattern48/Pattern48.java
import java.util.Scanner;

public class Pattern48 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the number of rows : ");
        int rows = sc.nextInt();
        printPattern(rows);

    }

    static void printPattern(int n) 
    { 
        int col, num = 0; 
 
        // loop for row
        for (int row = 1; row <= n; row++) { 
 
            // when row number is odd,then print in increasing order.
            if (row % 2 != 0) { 
 
                // printing in ascending order
                for (col = num + 1; col < num + row; col++) 
                    System.out.print(col + " "); 
                System.out.println(col++); 
 
                // update value of 'num' 
                num = col; 
            } 
 
         // when row number is odd,then print in descending order.
            else { 
 
                // update value of 'num' 
                num = num + row - 1; 
 
                // print numbers with the ' ' in 
                // decreasing order 
                for (col = num; col > num - row + 1; col--) 
                    System.out.print(col + " "); 
                System.out.println(col); 
            } 
        } 
    }
}